token/program-2022/src/extension/transfer_fee/mod.rs

```diff
@@ -49,6 +49,46 @@ impl TransferFee {
             Some(cmp::min(raw_fee, u64::from(self.maximum_fee)))
         }
     }
+
+    /// Calculate the fee that would produce the given output
+    pub fn calculate_inverse_fee(&self, post_fee_amount: u64) -> Option<u64> {
```

Comment:
ok it took me some notes to self and some review of https://github.com/solana-labs/solana-program-library/pull/3523 to understand the use case. the concept here is... transfer fees are deducted from the total credit to the recipient, not added to the total debit from the sender. the inverse fee is working backwards from a net amount, to determine the gross. and this is used in token-swap to take the amount the pool received and calculate the amount that must have been taken from the user. right?

Reply:
Yes exactly! And I was thinking that this could be useful more generally, similar to payment platforms that allow you to pay transfer fees for the recipient so they receive an exact amount.